### PR TITLE
FIX: CI: avoid passing Cython files to ruff

### PR DESCRIPTION
**Pull request by mckib2**
#### Reference issue

- resolves CI lint issues discovered in https://github.com/scipy/scipy/pull/17961

#### What does this implement/fix?

- avoids passing unsupported Cython files to Ruff linter

### CODE REVIEW

File: tools/lint.py [hunk 1]
```diff
@@ -116,9 +116,10 @@ def main():
     else:
         files = args.files
 
-    cython_exts = ['.pyx']  # Add '.pxd', '.pxi' once cython-lint supports it
-    cython_files = [f for f in files if any(f.endswith(ext) for ext in cython_exts)]
-    other_files = [f for f in files if not f.endswith('.pyx')]
+    cython_exts = ('.pyx', '.pxd', '.pxi')
+    # Add '.pxd', '.pxi' once cython-lint supports it
+    cython_files = [f for f in files if f.endswith('.pyx')]
```
Comment (stefanv) [inline tools/lint.py:121]: The Cython linter supports these files now, so I'd leave this line as it was and fix up the few errors reported, if any.

Comment (mckib2) [reply tools/lint.py]: Updated to include `pxd` and `pxi` files

File: tools/lint.py [hunk 2]
```diff
@@ -116,9 +116,9 @@ def main():
     else:
         files = args.files
 
-    cython_exts = ['.pyx']  # Add '.pxd', '.pxi' once cython-lint supports it
+    cython_exts = ('.pyx', '.pxd', '.pxi')
     cython_files = [f for f in files if any(f.endswith(ext) for ext in cython_exts)]
-    other_files = [f for f in files if not f.endswith('.pyx')]
+    other_files = [f for f in files if not any(f.endswith(ext) for ext in cython_exts)]
```
Comment (stefanv) [inline tools/lint.py:121]: Looks good, thanks! Another alternative would be to use set logic.

Comment (mckib2) [reply tools/lint.py]: Just pushed -- did you have something like that in mind?

File: tools/lint.py [hunk 3]
```diff
@@ -71,7 +71,7 @@ def run_ruff(files, fix):
         return 0, ""
     args = ['--fix'] if fix else []
     res = subprocess.run(
-        ['ruff', f'--config={CONFIG}'] + args + files,
+        ['ruff', f'--config={CONFIG}'] + args + list(files),
```
Comment (stefanv) [inline tools/lint.py:74]: Isn't files already guaranteed to be a list?

File: tools/lint.py (hunk 2 above)
Comment (stefanv) [reply tools/lint.py]: Thanks, yes, that's it!

File: tools/lint.py (hunk 3 above)
Comment (mckib2) [reply tools/lint.py:74]: Not if sets are used below -- `files` here is the result of a set difference

**Review by stefanv — APPROVED** [review]

**Comment by stefanv** [comment]
I took the liberty of resolving the merge conflict; should be good now. I ran it locally and saw no linting errors. Thanks @mckib2!

**Pull request merged**